app/src/main/kotlin/com/wire/android/ui/home/messagecomposer/attachment/AttachmentOptions.kt

```diff
@@ -48,22 +71,22 @@ fun AttachmentOptionsComponent() {
 }
 
 @Composable
-private fun FileBrowserFlow() = rememberOpenFileBrowserFlow(
-    onFileBrowserItemPicked = { pickedFileUri ->
-        // TODO: call vm to share raw file data
-        appLogger.d("pickedUri is $pickedFileUri")
-    },
-    onPermissionDenied = { /* TODO: Implement denied permission rationale */ }
-)
+private fun FileBrowserFlow(viewModel: AttachmentOptionsViewModel): UseStorageRequestFlow {
+    val context = LocalContext.current
+    return rememberOpenFileBrowserFlow(
+        onFileBrowserItemPicked = { pickedFileUri -> viewModel.prepareAttachment(context, pickedFileUri) },
+        onPermissionDenied = { /* TODO: Implement denied permission rationale */ }
+    )
+}
 
 @Composable
-private fun GalleryFlow() = rememberOpenGalleryFlow(
-    onGalleryItemPicked = { pickedPictureUri ->
-        // TODO: call vm to share raw pic data
-        appLogger.d("pickedUri is $pickedPictureUri")
-    },
-    onPermissionDenied = { /* TODO: Implement denied permission rationale */ }
-)
+private fun GalleryFlow(viewModel: AttachmentOptionsViewModel): UseStorageRequestFlow {
+    val context = LocalContext.current
+    return rememberOpenGalleryFlow(
+        onGalleryItemPicked = { pickedPictureUri -> viewModel.prepareAttachment(context, pickedPictureUri) },
+        onPermissionDenied = { /* TODO: Implement denied permission rationale */ }
+    )
+}
```

Comment:
I'm not a huge fan of passing any `context` instance to view models as they are heavy objects, and viewmodels lifecycle may be too long sometimes. Maybe we could just do all the transformations from Uri to ByteArray within the bounds of the composable item (or even better extract it to other utils method) and just pass the attachment raw data to the view model?

Reply:
Totally agree! Just took a shortcut seeing other viewmodel using android components, but we have to start moving away from this approach to start testing as you mentioned :D 

Anyway I'm refactoring this and moving away from the viewmodel approach, so I think this will be outdated code